pagopa/io-app
# fix: [IOBP-446] Add gallery permission request in barcode scan screen for Android devices

## Short description
This PR adds the `READ_MEDIA_IMAGES` permissions system dialog on Android and the permission instructions screen.

## List of changes proposed in this pull request
- [ts/utils/permission.ts](https://github.com/pagopa/io-app/pull/5313/files#diff-7752f292bcc12a6ee19233df702f1e1bafc087f6261759e0a5939c6e7a883f19): added wrapper function to request media permissions
- Added `AndroidMediaPermissionRequestScreen` to display instructions on how to grant permissions
- Added required i18n keys
- Added patches for `react-native+0.69.9` and `@types/react-native+0.69.9` in order to support READ_MEDIA_IMAGES permission on Android API Level 33+

## How to test
With a fresh app install on an Android device:
- Navigate to the barcode scan screen and check that the upload function displays the permission prompt, on both Android API level below **33** and **33+**

## Preview

| Android 11 | Android 14 |
| --- | --- |
| <video src="https://github.com/pagopa/io-app/assets/6160324/76ce92f6-0f7a-4d59-885b-19f424202f8b" width="200" /> | <video src="https://github.com/pagopa/io-app/assets/6160324/beaa673b-1fad-4002-8fb5-091efc8a11c4" width="200" /> |
